libraries/AP_Notify/NCP5623.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+  NCP5623 I2C LED driver
+*/
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "NCP5623.h"
+#include <utility>
+
+#include <AP_HAL/AP_HAL.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define NCP5623_LED_BRIGHT  0x1f    // full brightness
+#define NCP5623_LED_MEDIUM  0x18    // medium brightness
+#define NCP5623_LED_DIM     0x0f    // dim
+#define NCP5623_LED_OFF     0x00    // off
+
+#define NCP5623_LED_I2C_ADDR 0x38    // default I2C bus address
+
+#define NCP5623_LED_PWM0    0x40    // pwm0 register
+#define NCP5623_LED_PWM1    0x60    // pwm1 register
+#define NCP5623_LED_PWM2    0x80    // pwm2 register
+#define NCP5623_LED_ENABLE  0x20    // enable register
+
+NCP5623::NCP5623(uint8_t bus)
+    : RGBLed(NCP5623_LED_OFF, NCP5623_LED_BRIGHT, NCP5623_LED_MEDIUM, NCP5623_LED_DIM)
+    , _bus(bus)
+{
+}
+
+bool NCP5623::write(uint8_t reg, uint8_t data)
+{
+	uint8_t msg[1] = { 0x00 };
+    msg[0] = ((reg & 0xe0) | (data & 0x1f));
+    bool ret = _dev->transfer(msg, 1, nullptr, 0);
+	return ret;
+}
+
+bool NCP5623::writes(uint8_t *data, uint8_t len)
+{
+    bool ret = 0;
+    uint8_t reg = data[0];
+    for (uint8_t i = 0; i < (len - 1); i++) {
+        ret = write(reg, data[i + 1]);
```

Comment:
Does the data sheet say the device supports auto-increment for register setting?  That would allow us to do less on the bus.